src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -5426,6 +5426,17 @@ BOOL IsSafeToCallExecutionManager()
 
 BOOL IsSafeToHandleHardwareException(PCONTEXT contextRecord, PEXCEPTION_RECORD exceptionRecord)
 {
+#ifdef FEATURE_EMULATE_SINGLESTEP    
+    Thread *pThread = GetThreadNULLOk();
+    if (pThread && pThread->IsSingleStepEnabled() &&
+        exceptionRecord->ExceptionCode != STATUS_BREAKPOINT &&
+        exceptionRecord->ExceptionCode != STATUS_SINGLE_STEP &&
+        exceptionRecord->ExceptionCode != STATUS_STACK_OVERFLOW)
+    {
+        pThread->HandleSingleStep(contextRecord, exceptionRecord->ExceptionCode);
```

Comment:
The updated version of the change looks like it still calls HandleSingleStep from two different places, once here and once over in [HandleHardwareException](https://github.com/dotnet/runtime/blob/cc2f60147dece0797d19f4aa6a12c44201ecfee4/src/coreclr/vm/exceptionhandling.cpp#L5606).

I was suggesting that we converge those into a single call here. Is there a reason we need two different calls? If yes it would be great to add a comment saying why we have both, and if not it would be nice to get rid of the HandleHardwareException portion and do it all here.